Automattic/woocommerce-payments
# Add preview pane to auto currency switch feature

Fixes #2523

#### Changes proposed in this Pull Request

This PR adds a preview link to Store settings section of Multi-currency settings page, which enables the user to preview how the automatic currency switch feature will affect the customer. 

**Note:** To get this work, merchant must have the local currency enabled already, because it won't show a auto currency change notice under the page when the currency isn't available. 

The preview modal is extracted from MC onboarding to its own component and rendered on the store settings section in a placeholder that's provided in the `ìncludes/multi-currency/Settings.php` settings page constructor array.

#### Screenshots:

##### The link:

![image](https://user-images.githubusercontent.com/3295/133833304-0fd9d7aa-94aa-4e9c-95c2-eeef0dd015f0.png)

##### The modal:

![image](https://user-images.githubusercontent.com/3295/133500494-be126314-b6d1-4853-af3c-4ff348766b2c.png)









#### Testing instructions





* Open *WooCommerce > Settings > Multi-Currency*
* Add some currencies if the list is empty to get the store settings section displayed.
* Make sure your list doesn't contain your local currency at first
* Check if the preview link is visible in the help text of Auto currency switch checkbox
* Click the link and see the notice isn't there.
* Add your currency to the list now
* Click the link and see the notice is visible, and showing your local currency is effective, check the prices are displayed in your local currency.

-------------------


**Post merge**
